src/Umbraco.Web.UI/Umbraco/Views/Preview/Index.cshtml

```diff
@@ -1,32 +1,33 @@
-﻿<!DOCTYPE html>
+﻿@inherits System.Web.Mvc.WebViewPage
+@{
+    var disableDevicePreview = ViewData["DisableDevicePreview"].ToString().ToLowerInvariant();
+    var extendPreviewHtml = ViewData["ExtendPreviewHtml"] != null ? ViewData["ExtendPreviewHtml"].ToString() : null;
+}
+<!DOCTYPE html>
 <html lang="en">
 <head>
     <title>Umbraco Canvas Designer</title>
     <link href="../assets/css/canvasdesigner.css" type="text/css" rel="stylesheet" />
     <link href="../lib/spectrum/spectrum.css" type="text/css" rel="stylesheet" />
     <link href="../lib/jquery-ui/jquery-ui-1.10.4.custom.min.css" type="text/css" rel="stylesheet" />
 </head>
-
 <body id="canvasdesignerPanel" ng-mouseover="outlinePositionHide()" ng-class="{ leftOpen: (showStyleEditor || showPalettePicker) && !showDevicesPreview }" ng-controller="Umbraco.canvasdesignerController">
-
+    @if (extendPreviewHtml != null)
+    {
+        @Html.Raw(extendPreviewHtml)
```

Comment:
@clausjensen I feel like being able to specify a PartialView path for this would make a lot more sense than just specifying a string. As I'm updating a few things will headless today i'll see about that option, seems like it would be much more flexible.